src/microservices/uk.ac.dundee.hic.nerd/src/uk/ac/dundee/hic/nerd/Program.java

```diff
@@ -0,0 +1,116 @@
+package uk.ac.dundee.hic.nerd;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.zip.GZIPInputStream;
+
+import edu.stanford.nlp.ie.crf.CRFClassifier;
+import edu.stanford.nlp.ling.CoreAnnotations;
+import edu.stanford.nlp.ling.CoreLabel;
+
+public class Program {
+	private final ServerSocket listener;
+	private final CRFClassifier<CoreLabel> c;
+	private final HashSet<String> classifications=new HashSet<String>(Arrays.asList(new String[] {"PERSON","LOCATION","ORGANIZATION"}));
+	private volatile static boolean shutdown=false;
+	
+	public void shutdown() {
+		try {
+			listener.close();
+		} catch (IOException e) {
+			// Ignore exceptions on close(), can't do anything about that anyway
+		}
+		shutdown=true;
+	}
+
+	Program() throws IOException, ClassCastException, ClassNotFoundException {
+		this.listener = new ServerSocket(1881,255,InetAddress.getByName("127.0.0.1"));
```

Comment:
Can we make this configurable in the config file? We may want to run across multiple hosts.

Reply:
Resolved offline